sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -129,18 +185,14 @@
   @GuardedBy("lock")
   private final TreeSet<SegmentId> segmentsNeedingRefresh = new TreeSet<>(SEGMENT_ORDER);
 
-  // Escalator, so we can attach an authentication result to queries we generate.
-  private final Escalator escalator;
-
   @GuardedBy("lock")
   private boolean refreshImmediately = false;
-  @GuardedBy("lock")
-  private long lastRefresh = 0L;
-  @GuardedBy("lock")
-  private long lastFailure = 0L;
+
   @GuardedBy("lock")
   private boolean isServerViewInitialized = false;
 
+  private int totalSegments = 0;
```

Comment:
At first I was wondering if this be `volatile`, but I don't think it matters since it is only called whenever sys segments scan is run, and not in a loop or anything where it would probably matter. Maybe we should add a comment that it is ok to be neither guarded by, volatile, or a concurrent type?

Reply:
Sure, added.